### ISSUE TITLE
Syntax issues with negative literals and infix types

### ISSUE THREAD

**Issue by soronpo** [lampepfl/dotty#2783]
```scala
type ~~[A, B]
type good = 2 ~~ 2
type bad = 2 ~~ -2 //Error:(9, 20) ';' expected but integer literal found.
type work_around = 2 ~~ (-2) // ')' expected, but integer literal found
```

Scalac has the same issue, so I'm wondering if this is the proper behavior.
However, the `work_around` works in Scalac.
https://github.com/typelevel/scala/issues/157

**Comment by odersky** [comment]
According to the Dotty grammar in 

    https://github.com/lampepfl/dotty/blob/master/docs/docs/internals/syntax.md

`-2` are two tokens, a prefix operator `-` and an integer literal `2`. So this is as specified.

**Comment by smarter** [comment]
But if we want to be compatible with scalac once it gets literal singleton types, we'll have to change our syntax, this works in https://github.com/scala/scala/pull/5310 but not in Dotty:
```scala
val x: -2 = -2
```
This is especially problematic because it means that there is no way to write some inferred types by hand in Dotty:
```scala
scala> type Id[T] = T
// defined alias type Id = [T] => T
scala> def narrow[T <: Singleton](x: T): Id[T] = x 
def narrow[T <: Singleton](x: T): Id[T]
scala> narrow(-2) 
val res0: Id[-2] = -2
scala> val x: Id[-2] = -2 
1 |val x: Id[-2] = -2
  |           ^
  |           an identifier expected, but integer literal found
```

**Comment by soronpo** [comment]
I don't know exactly when this issue was resolved, but in v0.20.0-RC1 dotty it works OK.
